runtime/vm/jnicsup.cpp

```diff
@@ -498,31 +501,23 @@ UDATA JNICALL   pushArguments(J9VMThread *vmThread, J9Method* method, void *args
 				*(I_32*)(--sp) = (I_32)(jint)ARG(jint, i);	/* NOTE: jint may be larger than int, but jchar, jshort, etc, must be no larger than int */
 				break;
 #ifdef J9VM_INTERP_FLOAT_SUPPORT
-			case 'F':
+			case 'F': {
 				/* float type */
-				fltPtr = (jfloat*)--sp;
-				/* The Linux SH4 compiler needs the next two lines to be two steps.  If you combine them it fails to compile */
-				dbl = ARG(jdouble, f);
-				*fltPtr = (jfloat)dbl;
-				break;
+				sp -= 1;
+				/* jfloat {aka float}' is promoted to 'double' when passed through '...' */
+				*((jfloat*) sp) = ARG(jdouble, f);
```

Comment:
My mistake = this does need a jfloat cast for the varargs case (which means always cast here).